db/commitlog/commitlog.cc

```diff
@@ -718,42 +757,30 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
                 return cycle(true, true);
             }
         }
-        return make_ready_future<sseg_ptr>(shared_from_this());
-    }
-    future<sseg_ptr> close() {
-        _closed = true;
-        auto s = co_await sync();
-        co_await flush();
-        co_await terminate();
-        _segment_manager->totals.wasted_size_on_disk += (_size_on_disk - file_position());
-        co_return s;
+        return make_ready_future<>();
     }
-    future<sseg_ptr> do_flush(uint64_t pos) {
-        auto me = shared_from_this();
-        co_await begin_flush();
-
-        auto finally = defer([&] () noexcept {
-            end_flush();
-        });
-
-        if (pos <= _flush_pos) {
-            clogger.trace("{} already synced! ({} < {})", *this, pos, _flush_pos);
-            co_return me;
+    void close() {
```

Comment:
I'd would rename this to e.g. `request_close` or `trigger_close`.

Reply:
Fair enough.

Reply:
fixed